### PR TITLE
Change/possession camera

### PR DESCRIPTION
**Pull request by KiwiLT**
## Description
You can now change from possession to possession. I had to add extra camera's for a smooth transition. Also unpossess the current object when repossessing (but not with changing to the player in-between).

## Setting up testing environment
1. In the project window navigate to Project/Assets/Scenes and open the "PossessionCamera" scene.
2. Press Play.

#### Aim mode offset
Criteria:
- [x] Releasing LMB when in Possess aim mode when hovering an object causes you to possess a new object.
- [x] The previous object is correctly unpossessed when repossessing.
- [x] The camera has a smooth transition.
- [x] Possession should still work correctly when repossessing, going back to player and then possessing again.

## Checklist
_These checks should always be true for your pull request_
- [x] My pull request is for one story/feature.
- [x] Every individual commit in this pull request is logical.
- [x] All code, documentation, commits and player seen texts are in English.
- [x] I have deleted unused / unnecessary code.
- [x] If my edits need a change in documentation, then I have updated the documentation.
- [x] All code is in correspondence with the code conventions.

### CODE REVIEW

**Comment by KiwiLT** [comment]
This is a cool feature but there might be pull requests that are more urgent. Check this before reviewing.

File: Assets/Scripts/Possession/PossessionController.cs [hunk 1]
```diff
@@ -51,15 +51,21 @@ public void Unpossess()
...
             _thirdPersonController.toUnpossessLocation();
-            CurrentPossession = null;
-            CurrentThrowable = null;
+            RemovePossession();
             _thirdPersonController.freeze = false;
             CurrentPossessionChanged?.Invoke();
         }
     }
 
+    private void RemovePossession()
```
Comment (Jeroenvn) [inline Assets/Scripts/Possession/PossessionController.cs:61]: the name of the function makes it sound like it does the same thing as Unpossess()

File: Assets/Scripts/Player/AimMode.cs [hunk 2]
```diff
@@ -144,16 +147,33 @@ public void ThrowObject()
...
+        if (_possessionDefaultCam == _cameras[2])
+        {
+            newDefaultCamera = _possessionDefaultCam1;
+            newAimCamera = _possessionAimCam1;
+
+        } else
+        {
+            newDefaultCamera = _possessionDefaultCam;
+            newAimCamera = _possessionAimCam;
+        }
```
Comment (Jeroenvn) [inline Assets/Scripts/Player/AimMode.cs:172]: This is difficult to read without context. This operation could be described in a short sentence SwapPossessionCamera or something along those lines. If you put this in a function with a name like that and call it it makes the code that much more readable.

File: Assets/Scripts/Player/AimMode.cs [hunk 3]
```diff
@@ -144,16 +147,33 @@ public void ThrowObject()
 
     public void changeCameraToPossession()
     {
-        ExitAimMode();
         if (_possessionController.CurrentPossession == null)
         {
+            ExitAimMode();
             return;
         }
         Transform pos = _possessionController.CurrentPossession.GetComponent<ClutterCamera>().CinemachineCameraTarget.transform;
```
Comment (Jeroenvn) [inline Assets/Scripts/Player/AimMode.cs]: Also please change the name pos to something more descriptive. pos is used for position.

File: Assets/Scripts/Player/AimMode.cs [hunk 4]
```diff
@@ -144,16 +147,33 @@ public void ThrowObject()
 
     public void changeCameraToPossession()
```
Comment (Jeroenvn) [inline Assets/Scripts/Player/AimMode.cs]: Methods should be camel case

Comment (Jeroenvn) [reply Assets/Scripts/Player/AimMode.cs]: same goes for switch camera

**Review by Jeroenvn — CHANGES_REQUESTED** [review]
It works! but there's some things in the code that could use some adjustments

**Review by Jeroenvn — APPROVED** [review]
Looks good!

**Pull request closed without merging**